### PR TITLE
Pricing Grid: Implement storage options display

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Proposed Changes

* This PR is one among several stacked PRs that implements the new pricing grid design. In https://github.com/Automattic/wp-calypso/pull/71372, we introduced a new signup flow /start/onboarding-2023-pricing-grid that is only available in dev and wpcalypso environments.
* The design for the pricing grid can be found in 2e991-pb. Please keep the Figma design open in a separate tab.
* The parent PR is https://github.com/Automattic/wp-calypso/pull/71511.
* This PR implements the Storage row in the features table.

Screenshot 2022-12-26 at 3 53 37 PM

#### What Not To Test

* Responsive screens. It will be implemented in a separate PR. Right now, test in a full width desktop browser viewport.
* Non-en. The first launch will be `en` only.

#### Testing Instructions

*  Navigate to `/start/onboarding-2023-pricing-grid`.
* Check that the storage options are displayed as per the design in Figma. 
* Verify that the production `onboarding` signup flow at `/start` is not affected by these changes.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/index.jsx [hunk 1]
```diff
@@ -12,9 +12,13 @@ import {
...
 	isEcommercePlan,
 	PLAN_ENTERPRISE_GRID_WPCOM,
+	FEATURE_1GB_STORAGE,
+	FEATURE_6GB_STORAGE,
+	FEATURE_13GB_STORAGE,
+	FEATURE_200GB_STORAGE,
 } from '@automattic/calypso-products';
 import classNames from 'classnames';
-import { localize } from 'i18n-calypso';
+import { localize, translate } from 'i18n-calypso';
```
Comment (aneeshd16) [inline client/my-sites/plan-features-2023-grid/index.jsx]: The `localize` HOC already adds the `translate` method to the component props, and it is the [recommended approach](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/README.md#localize). Can you help me understand why we have imported the `translate` method here?

File: client/my-sites/plan-features-2023-grid/index.jsx [hunk 2]
```diff
@@ -258,6 +263,52 @@ export class PlanFeatures2023Grid extends Component {
 			<PlanFeatures2023GridFeatures planProperties={ planProperties } domainName={ domainName } />
 		);
 	}
+
+	getFeatureToStorageMap( storageFeature ) {
```
Comment (aneeshd16) [inline client/my-sites/plan-features-2023-grid/index.jsx]: Minor nit: This method doesn't return a map but returns a translated string. Can we maybe rename this to `getStorageStringFromFeature` or similar?

File: client/my-sites/plan-features-2023-grid/index.jsx [hunk 3]
```diff
@@ -258,6 +263,52 @@ export class PlanFeatures2023Grid extends Component {
...
+		switch ( storageFeature ) {
+			case FEATURE_1GB_STORAGE:
+				return translate( '1GB' );
+			case FEATURE_6GB_STORAGE:
+				return translate( '6GB' );
+			case FEATURE_13GB_STORAGE:
+				return translate( '13GB' );
+			case FEATURE_200GB_STORAGE:
+				return translate( '200GB' );
```
Comment (aneeshd16) [inline client/my-sites/plan-features-2023-grid/index.jsx]: Minor: In the design, the text is `200 GB` (there's a space between 200 & GB)
image

**Review by aneeshd16 — APPROVED** [review]
Overall LGTM, I left some minor feedback.

File: client/my-sites/plan-features-2023-grid/index.jsx (hunk 3 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/plan-features-2023-grid/index.jsx]: You are right about the design, but in the product we seem to be using both representations - `X GB` and `XGB`, e.g [this line](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/plans/features-list.js#L1324). For now I'll add the space, and I'll verify with the designer on the best approach to take.

File: client/my-sites/plan-features-2023-grid/index.jsx (hunk 2 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/plan-features-2023-grid/index.jsx]: Done, pushed a change for this.

File: client/my-sites/plan-features-2023-grid/index.jsx (hunk 1 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/plan-features-2023-grid/index.jsx]: Correct, we can remove this (done in a commit I just pushed)

**Pull request merged**